src/components/scene/background.js

```diff
@@ -1,49 +1,229 @@
-/* global THREE */
+/* global THREE, XRWebGLBinding, Map */
 var register = require('../../core/component').registerComponent;
 var COMPONENTS = require('../../core/component').components;
 
 module.exports.Component = register('background', {
   schema: {
-    color: {type: 'color', default: 'black'},
-    transparent: {default: false},
-    generateEnvironment: {default: true}
+    color: { type: 'color', default: 'black' },
+    transparent: { default: false },
+    generateEnvironment: { default: true },
+    environmentUpdateFrequency: { default: 0 },
+    sceneLights: { default: 'a-light,[light]' }
   },
   init: function () {
+    var self = this;
+
     this.cubeRenderTarget = new THREE.WebGLCubeRenderTarget(128, { format: THREE.RGBFormat, generateMipmaps: true, minFilter: THREE.LinearMipmapLinearFilter });
+    this.lightProbeTarget = new THREE.WebGLCubeRenderTarget(16, { format: THREE.RGBFormat, generateMipmaps: false });
     this.cubeCamera = new THREE.CubeCamera(1, 100000, this.cubeRenderTarget);
     this.needsEnvironmentUpdate = true;
+
+    this.el.sceneEl.addEventListener('enter-vr', function () {
+      var renderer = self.el.renderer;
+      var session = renderer.xr.getSession();
+      if (
+        session.requestLightProbe && self.el.sceneEl.is('ar-mode')
+      ) {
+        self.startLightProbe();
+      }
+    });
+
+    this.el.sceneEl.addEventListener('exit-vr', function () {
+      self.stopLightProbe();
+    });
+
+    var directionalLight = document.createElement('a-light');
```

Comment:
Should we make this conditional and only add this lights if / when background estimation is enabled to keep the scene clean?

Reply:
Maybe add a method `setupProbeLights` that it's invoked from the `startLightProbe` method. It would be no-op if the lights have been already added.

Reply:
Making a `setupProbeLights` method makes sense.

A lot of the setup is done in the tick function because the "enter-vr" event is often fired before the session is started.

I will probably move the things which are setup on the first tick after the session starts into a separate setupProbe method which will invoke setupProbeLights